source/read-data-from-mongo.txt

```diff
@@ -0,0 +1,188 @@
+.. _java-rs-read:
+
+======================
+Read Data From MongoDB
+======================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+ 
+.. meta::
+   :description: Learn how to use the Java Reactive Streams driver to read data from MongoDB.
+   :keywords: usage examples, see, crud, access, code example
+
+.. TODO: change toctree links when other pages are merged in.
+.. .. toctree::
+..    :titlesonly:
+..    :maxdepth: 1
+
+..    /read/specify-query
+..    /read/retrieve-data
+..    /read/specify-fields
+..    /read/specify-documents
+..    /read/count-documents
+
+Overview
+--------
+
+This page contains copyable code examples of {+driver-short+} methods that you can use to
+read data from MongoDB.
+
+.. tip::
+
+  To learn more about any of the methods shown on this page, see the link
+  provided in each section.
+
+To use an example from this page, copy the code example into the
+:ref:`sample application <java-rs-read-sample>` or your own application.
+Be sure to replace all placeholders in the code examples, such as ``<connection string>``, with
+the relevant values for your MongoDB deployment.
+
+Project Reactor Implementation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This guide uses the Project Reactor library to consume ``Publisher`` instances returned
```

Comment:
S: can you use the source constant here for Project Reactor? Applies wherever used